Source/ARTPaginatedResult.m

```diff
@@ -67,7 +83,30 @@ - (instancetype)initWithItems:(NSArray *)items
     return self;
 }
 
+- (void)initializedViaInitCheck {
+    if (_initializedViaInit) {
+        [NSException raise:NSInvalidArgumentException format:@"When initializing this class using -init, you need to override this method in a subclass"];
```

Comment:
```suggestion
        [NSException raise:NSInternalInconsistencyException format:@"When initializing this class using -init, you need to override this method in a subclass"];
```

(I think I wrote the `NSInvalidArgumentException` but actually this seems more appropriate.)

Reply:
done